### PR TITLE
Chore: Adding log also for cases where datasource UID length is invalid

### LINKED ISSUE #84442: Extend logging to also include UID length check

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What is this feature?**
Extending invalid UID logging to also include cases where UID length is invalid. This PR follows https://github.com/grafana/grafana/issues/68447

**Why do we need this feature?**

Makes logging for invalid UID cases complete.

**Who is this feature for?**
Preparation for enforcing UID format in the future.

**Which issue(s) does this PR fix?**:

Fixes #84442

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by xnyo — APPROVED** [review]
LGTM!

File: pkg/services/datasources/service/store.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func (ss *SqlStore) AddDataSource(ctx context.Context, cmd *datasources.AddDataS
 				return fmt.Errorf("failed to generate UID for datasource %q: %w", cmd.Name, err)
 			}
 			cmd.UID = uid
-		} else if !util.IsValidShortUID(cmd.UID) {
+		} else if err := util.ValidateUID(cmd.UID); err != nil {
```
Comment (marefr) [inline pkg/services/datasources/service/store.go:256]: Maybe we should include the error in the log as well?

Comment (tolzhabayev) [reply pkg/services/datasources/service/store.go:256]: Good point, added

**Review by marefr — APPROVED** [review]
LGTM

**Comment by blockchainchangeworld** [comment]
NB

**Pull request merged**